conantools.cmake

```diff
@@ -128,7 +140,7 @@ function(conan_provide_dependency package_name)
     if(NOT CONAN_INSTALL_SUCCESS)
         message(STATUS "CMake-conan: first find_package() found, running 'conan install' to install deps")
         detect_host_profile(${CMAKE_BINARY_DIR}/conan_host_profile)
-        if(CMAKE_BUILD_TYPE)
+        if(NOT CMAKE_CONFIGURATION_TYPES)
```

Comment:
I was relying on the fact that users in Windows can still force installing just 1 configuration with ``-DCMAKE_BUILD_TYPE=Debug``, even if that does nothing, Conan will know to install only the Debug.

If we are not allowing this, we would need another kind of opt-out I guess?

Reply:
yup, we would need some opt-out indeed!  I would try to avoid giving `CMAKE_BUILD_TYPE` a meaning in the context of a multi-config generator (where giving `CMAKE_BUILD_TYPE` a value at configure time is unfortunately allowed, but will have no effect).